controller/cmd/tap/main.go

```diff
@@ -45,17 +49,46 @@ func main() {
 		log.Fatal(err.Error())
 	}
 
+	_, port, err := net.SplitHostPort(lis.Addr().String())
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	// TODO: remove the network hop in favor of APIServer calling TapByResource
+	// directly.
+	tapClient, cc, err := tap.NewClient("127.0.0.1:" + port)
```

Comment:
This is interesting. Why does there need to be this hop here introducing an "intermediate" client?

Reply:
It's not strictly necessary, was just a quick and dirty way to wire up the APIServer to the Tap Controller.

Reply:
yup, filed #3171 to track, and linked to this comment.